### PR TITLE
frontend: Cache all but HTML

### LINKED ISSUE #920: frontend often tries to load spinninglogo after backend is down, so it doesn't show at all
image

### PR DESCRIPTION
**Pull request by rafaellehmkuhl**
Cache all files apart from HTML documents.
Only HTML documents do not receive an SHA and need to be re-fetched to guarantee we update assets correctly on version-updates.

Fix #920

Image available at `rafaellehmkuhl/blueos-core`

### CODE REVIEW

**Comment by Williangalvani** [comment]
well, it is cached!
image

File: core/tools/nginx/nginx.conf [hunk 1]
```diff
@@ -164,8 +164,10 @@ http {
         location / {
             root /home/pi/frontend;
             try_files $uri $uri/ /index.html;
+        }
+
+        location ~* /.(html|css|js)$ {
             expires -1;
-            add_header Cache-Control no-store;
```
Comment (Williangalvani) [inline core/tools/nginx/nginx.conf:168]: I suspect we need this. it seems to be caching files it shouldn't:
image

Comment (rafaellehmkuhl) [reply core/tools/nginx/nginx.conf:168]: Good catch. Thought the "expires" was enough. Rebuilding.

**Review by Williangalvani — CHANGES_REQUESTED** [review]
I believe you can drop the first commit now,
there's also a whitespace change, but I like this approach.

**Comment by Williangalvani** [comment]
I can't get this to cache at all. Am I doing something wrong?

**Review by Williangalvani — APPROVED** [review]
Nevermind. looks good!
image

**Pull request merged**